### ISSUE TITLE
Add labels to help classify what is needed for open needsdiagnosis to progress

### ISSUE THREAD

**Issue by wisniewskit** [webcompat/webcompat.com#1413]
Since we always have needsdiagnosis bugs which are waiting on external information or fixes, I'd like to propose some potential new labels:

- need-regional-testing-[country]
- need-device-[model]
- need-login-information
- pending-browser-fix
- pending-external-diagnosis
- pending-gofaster-update
- pending-spec-work

This way contributors can more easily find the bugs for which they can help to complete the diagnosis, and we have an easier time knowing why a given bug is still needsdiagnosis.

**Comment by karlcow** [comment]
Thanks!

so some of the proposed labels here are in part what we are discussing in #1339
We figure out that everything which requires `label-some_variable` is doomed to increase a lot.

## Benefits of using labels
* easily [searchable on github](https://developer.github.com/v3/issues/labels/).
* easier to not make typo, aka select-able instead of typing (controlled vocabulary)

## Cons of using labels
* Proliferation of keywords will make the list unusable with the current UI. 

I wonder if the UI for editing labels will become something like stackoverflow or even github projects keywords. 
To think. 

@wisniewskit Could you describe a bit more each keyword here? I think I get the meaning for all of them, but for the future and new contributors, or people not completely aware of how the project it is working.

**Comment by wisniewskit** [comment]
>We figure out that everything which requires label-some_variable is doomed to increase a lot.

I agree, but I suspect that this might be manageable if the labels aren't "permanent", but just there to mark in-progress statuses (as in, once they don't apply we can remove them, and prune them if they're not longer being used).

>Could you describe a bit more each keyword here?

Certainly:
- need-regional-testing-[country]: for issues on which we are stuck waiting for help from someone from a specific region/country, because the site in question is region-locked.
- need-device-[model]: for issues where we are stuck because we don't have the phone/tablet/etc necessary to reproduce the issue, and are waiting for help from someone with one of those devices.
- need-login-information: for issues where a web site/service problem cannot be diagnosed without having a login, which we do not have (or if a fresh/free account does not exhibit the issue).
- pending-browser-fix: for when we have partly diagnosed an issue, and must wait until a browser vendor fixes a bug/etc before we can be sure that the issue is truly fixed.
- pending-external-diagnosis: for issues where we have deferred to other experts (like web-specification writers or browser engineers) to help us complete our diagnosis, and are waiting for their input.
- pending-gofaster-update: for issues we believe can be solved using the gofaster Firefox addon, but are not fully certain that further diagnosis is necessary.
- pending-spec-work: similarly, for issues where we know an issue is related to browsers behaving differently because the relevant web standards are flawed (or don't yet exist). It should be useful to mark them with this label until the relevant issues against the specification are all filed.

**Comment by karlcow** [comment]
>I agree, but I suspect that this might be manageable if the labels aren't "permanent", but just there to mark in-progress statuses (as in, once they don't apply we can remove them, and prune them if they're not longer being used).

Two aspects of it.
* labels are "permanent" in the drop down list. That's my doomed to "increase a lot". For example, the `need-regional-testing-[country]` could become long. On Tech Evangelism we have the whiteboard keyword `country-fr`, `country-jp`, `country-ca` for the location of the site and `country-all` when it's basically everywhere. While we will probably not have [206 countries/labels](https://en.wikipedia.org/wiki/List_of_sovereign_states), this gives the upper limit for this tag only.

Thanks a lot for the description it helps!

* `pending-browser-fix` is basically #1339
* `pending-external-diagnosis`. Here there is something probably more generic which is maybe related to assignment or ping. It would be useful to have a "person blocker" like we have bug dependencies. And that should be useful in between us too. On Bugzilla this is managed by a softer version of a blocker with needinfo. We have a status needinfo that we should kill I think and reshape it in a real needinfo.
* `pending-gofaster-update` this is very specific. And there is probably an interesting thing around that. `gofaster` is Mozilla only and a generic term not only for webcompat. So probably a better name for something related to browser override of some sort. Mozilla has `gofaster-webcompat`. Opera has `browserjs`. I wonder about other browsers. Sometimes things happen directly in the source code on Firefox too.
* `pending-spec-work` something ala `seeAlso` or do you want it be a blocker.

So all of these seems to be blocking status, aka I can't do the work because of reason X. It's good to know in terms of UI.

I have the feeling that at a point we will be using github as a messaging platform more than a issue repo.

**Comment by karlcow** [comment]
See also the discussion about prioritization in #715

**Comment by karlcow** [comment]
This is a subtopic of #997

**Comment by karlcow** [comment]
I just assigned @denschub and @wisniewskit to come up with a list of labels and so we can have a better understanding of how we can make all of this fit together with the #997 issue too.
